ape_tokens/managers.py

```diff
@@ -106,21 +108,32 @@
 )
 
 
-class TokenManager(ManagerAccessMixin, dict):
+class TokenManager(ManagerAccessMixin, Mapping[str, ContractInstance]):
     @cached_property
     def _manager(self) -> TokenListManager:
         return TokenListManager()
 
     def __repr__(self) -> str:
         return f"<ape_tokens.TokenManager default='{self._manager.default_tokenlist}'>"
 
-    def __getitem__(self, symbol: str) -> ContractInstance:
-        try:
-            token_info = self._manager.get_token_info(
-                symbol, chain_id=self.network_manager.network.chain_id
-            )
+    def __len__(self) -> int:
+        return len(list(self._manager.get_tokens(chain_id=self.provider.chain_id)))
 
-        except ValueError as err:
+    def __iter__(self) -> Iterator[ContractInstance]:
+        for token in self._manager.get_tokens(chain_id=self.provider.chain_id):
+            yield self[token.symbol]
+
+    def __getitem__(self, symbol: str) -> ContractInstance:
+        token_info = None
+        for tokenlist in self._manager.available_tokenlists():
+            try:
+                token_info = self._manager.get_token_info(
+                    symbol, chain_id=self.network_manager.network.chain_id, token_listname=tokenlist
+                )
+            except ValueError:
```

Comment:
else `err` is not defined below:

```suggestion
            except ValueError as err:
```